ipi/inputs/forcefields.py

```diff
@@ -831,3 +833,225 @@ def fetch(self):
             active_thresh=self.active_thresh.fetch(),
             active_out=self.active_output.fetch(),
         )
+
+
+class InputFFCavPhSocket(InputForceField):
+
+    """Creates a ForceField object with a socket interface for cavity molecular dynamics.
+
+    Handles generating one instance of a socket interface forcefield class.
+
+    Attributes:
+       mode: Describes whether the socket will be a unix or an internet socket.
+
+    Fields:
+       address: The server socket binding address.
+       port: The port number for the socket.
+       slots: The number of clients that can queue for connections at any one
+          time.
+       timeout: The number of seconds that the socket will wait before assuming
+          that the client code has died. If 0 there is no timeout.
+    """
+
+    fields = {
+        "address": (
+            InputValue,
+            {
+                "dtype": str,
+                "default": "localhost",
+                "help": "This gives the server address that the socket will run on.",
+            },
+        ),
+        "port": (
+            InputValue,
+            {
+                "dtype": int,
+                "default": 65535,
+                "help": "This gives the port number that defines the socket.",
+            },
+        ),
+        "slots": (
+            InputValue,
+            {
+                "dtype": int,
+                "default": 4,
+                "help": "This gives the number of client codes that can queue at any one time.",
+            },
+        ),
+        "exit_on_disconnect": (
+            InputValue,
+            {
+                "dtype": bool,
+                "default": False,
+                "help": "Determines if i-PI should quit when a client disconnects.",
+            },
+        ),
+        "timeout": (
+            InputValue,
+            {
+                "dtype": float,
+                "default": 0.0,
+                "help": "This gives the number of seconds before assuming a calculation has died. If 0 there is no timeout.",
+            },
+        ),
```

Comment:
I don't think this code repetition is needed here.
Look for example the InputFFDebye class.